lib/Linter.js

```diff
@@ -586,6 +589,35 @@ function stripUnicodeBOM(text) {
     return text;
 }
 
+/**
+ * Get the severity level of a rule (0 - none, 1 - warning, 2 - error)
+ * Returns 0 if the rule config is not valid (an Array or a number)
+ * @param {Array|number} ruleConfig rule configuration
+ * @returns {number} 0, 1, or 2, indicating rule severity
+ */
+function getRuleSeverity(ruleConfig) {
+    if (typeof ruleConfig === "number") {
```

Comment:
With this PR, i am not modifying any other aspects of the core. This is what we had.
To be honest, I don't want to do too much in a single PR, just want to keep it focused on the objective in hand.
With that said, if you think their is something not right here, we should open an issue and handle it separately. I hope this makes sense.